frontend/packages/dev-console/src/components/topology/TopologyEdgePanel.tsx

```diff
@@ -49,10 +49,11 @@ const connectorTypeToTitle = (type: string): string => {
 };
 
 const TopologyEdgePanel: React.FC<TopologyEdgePanelProps> = ({ edge, data, consoleLinks }) => {
-  const source: TopologyDataObject = edge.getSource().getData();
-  const target: TopologyDataObject = edge.getTarget().getData();
+  const panelEdge = edge.getChildren()?.length === 1 ? (edge.getChildren()[0] as Edge) : edge;
```

Comment:
Instead of checking `edge.getChildren()?.length === 1` could we be specific and check the edge type for the aggregate edge?

Reply:
The type for aggregate edges depicting a single edge is the original edge type allowing it to create the correct Edge element for display purposes.